rtrlib/lib/alloc_utils.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * This file is part of RTRlib.
+ *
+ * This file is subject to the terms and conditions of the MIT license.
+ * See the file LICENSE in the top level directory for more details.
+ *
+ * Website: http://rtrlib.realmv6.org/
+ */
+
+#ifndef LRTR_ALLOC_UTILS_H
+#define LRTR_ALLOC_UTILS_H
+
+#include <stdlib.h>
+
+extern void *(*MALLOC_PTR)(size_t size);
+extern void (*FREE_PTR)(void *ptr);
+extern void *(*REALLOC_PTR)(void *ptr, size_t size);
+
+/**
+ * @brief Sets a custom malloc function that is used throughout rtrlib.
+ * If you set this function you must also set custom free and custom realloc
+ * @param[in] Pointer to a malloc function
+ */
+void lrtr_set_malloc(void *(*malloc_function)(size_t size));
```

Comment:
If always all functions has to be set, an alternative could be to have a single register() function that takes 3 function pointers or a struct that contains the function pointers.
I don't know other libraries implementing it, maybe there is a common pattern for it that we can follow (if we are not doing it already).